src/Checkbox.js

```diff
@@ -1,175 +1,192 @@
 import cx from 'classnames'
 import propTypes from '@dhis2/prop-types'
-import React, { Component, Fragment } from 'react'
+import React, { PureComponent, createRef } from 'react'
```

Comment:
One thing that I've noticed in ui-core is the PureComponents we're using in some places. I kind of agree with this article https://cdb.reacttraining.com/react-inline-functions-and-performance-bdff784f5578, I think we should be careful about optimizing at this stage of the lib.

Especially since we're not measuring anything to see whether it actually makes a difference, and it can cause subtle bugs.

Reply:
Good point, you're correct in thinking that this is a premature optimisation which is purely based on an assumption. We have quite a lot of components with a lot of prop, of which only a few are likely to change, but as the article you link to explains, that doesn't always mean stuff will get faster.. I think your suggestion is good: let's stick to `Component` for now, and only start optimising (i.e. using `PureComponent` etc) when we notice something.

Reply:
@ismay that's a very nice article you got there!

Reply:
@Mohammer5 Yeah I liked it as well, he makes a good point.